common/timeprofile.h

```diff
@@ -73,12 +73,12 @@ static inline void timeprofile_clear(timeprofile_t *tp)
     tp->utime = utime_now();
 }
 
+// Caller owns name memory, use static memory: timeprofile_stamp(tp, "profile-section"); 
 static inline void timeprofile_stamp(timeprofile_t *tp, const char *name)
 {
     struct timeprofile_entry tpe;
 
-    strncpy(tpe.name, name, sizeof(tpe.name));
-    tpe.name[sizeof(tpe.name)-1] = 0;
+    tpe.name = name;
```

Comment:
If the caller owns the memory, how do we make sure it is not free'd before `tp->stamps` is used?

Reply:
It is always static memory passed to an internal API e.g. the memory lives inside the .text segment of the library and has the same lifetime scope as the functions accessing it:

![image](https://user-images.githubusercontent.com/987907/155407380-8185ab06-21d2-406b-9e10-fb06b30efaa5.png)

Reply:
I agree with @christian-rauch. Even if currently there are no uses of timeprofile_stamp which would cause a problem, I'd rather not create a footgun.

Was the goal of this change to speed up detection? If so, I would be willing to accept this change if it gives a substantial speedup